src/providers/ogr/qgsgeopackagedataitems.cpp

```diff
@@ -370,22 +369,39 @@ QList<QAction *> QgsGeoPackageVectorLayerItem::actions()
 
 void QgsGeoPackageVectorLayerItem::deleteLayer()
 {
-  if ( QMessageBox::question( nullptr, QObject::tr( "Delete Layer" ),
-                              QObject::tr( "Are you sure you want to delete layer '%1' from GeoPackage?" ).arg( mName ),
-                              QMessageBox::Yes | QMessageBox::No, QMessageBox::No ) != QMessageBox::Yes )
-    return;
-
-  QString errCause;
-  bool res = ::deleteLayer( mUri, errCause );
-  if ( !res )
+  // Check if the layer is in the registry
+  const QgsMapLayer *projectLayer = nullptr;
+  Q_FOREACH ( const QgsMapLayer *layer, QgsProject::instance()->mapLayers() )
```

Comment:
@Gustry don't ask me :) I also thought it was to be avoided but it looks like everybody is still using it, random hits:
https://github.com/qgis/QGIS/commit/a62b02699d493648c349cc2d2906a55cb0afaa3a#diff-133f6835a6f0129f68962941ac54e0d5R112
https://github.com/qgis/QGIS/commit/e52414ea9dfd68fae3f6da9bd1205794abeb7ecf#diff-bc479904c7553f9ba44fd500dcd42eecR401

@m-kuhn @nyalldawson  any hint? If we reach a decision it would be good to spread the word on the dev ML and on the dev guidelines.